lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1474,6 +1492,11 @@ def cluster_name_from_kube_env(kube_env)
       instance_prefix
     end
 
+    def location_from_kube_env(kube_env)
+      # If 'ZONE' is not present return nil.
+      return kube_env['ZONE'] if kube_env.key?('ZONE')
```

Comment:
The metadata agent was written to also work in older clusters. This code is intended to work in newer clusters only (1.10+), as far as I know. The simpler we can make this code, the easier it would be to convince ourselves that it's correct. But I don't have a strong opinion about this except for the cleanup I've already suggested.

Reply:
Hmm... I'll prefer consistency with Metadata Agent. Mainly because as a user I would assume a fallback to have the same behavior. The cleanup is done.

Reply:
Users who will be affected by this fallback shouldn't see a difference, because new clusters define the metadata attribute, so this code would never be invoked. But I won't insist, as long as it's well encapsulated.